### PR TITLE
link: ipvlan: Change flag mode from u16 to Flag.

### PR DESCRIPTION
**Pull request by xujunjie-cover**
0 => bridge;
1 => private;
2 => vepa;

### CODE REVIEW

File: src/link/link_info/ipvlan.rs [hunk 1]
```diff
@@ -165,3 +167,43 @@ impl From<IpVlanMode> for u16 {
         }
     }
 }
+
+const IPVLAN_FLAG_BRIDGE: u16 = 0;
+const IPVLAN_FLAG_PRIVATE: u16 = 1;
```
Comment (cathay4t) [inline src/link/link_info/ipvlan.rs:172]: Please use kernel constant name to helping other developers to validate.

In this case, it should be `IPVLAN_F_PRIVATE`

**Review by cathay4t — CHANGES_REQUESTED** [review]
The `IpVlanFlag` is a bitflag. Please check kernel function `ipvlan_mark_vepa`.

Please check other code which is using `bitflags!` .

File: src/link/link_info/ipvlan.rs (hunk 1 above)
Comment (xujunjie-cover) [reply src/link/link_info/ipvlan.rs]: done. thanks

File: src/link/link_info/ipvlan.rs [hunk 2]
```diff
@@ -165,3 +163,24 @@ impl From<IpVlanMode> for u16 {
...
     }
 }
+
+const IPVLAN_F_PRIVATE: u16 = 0x01;
+const IPVLAN_F_VEPA: u16 = 0x02;
+
+bitflags! {
+    #[non_exhaustive]
+    #[derive(Debug, Clone, Copy, Eq, PartialEq)]
+    pub struct IpVlanFlag: u16 {
```
Comment (cathay4t) [inline src/link/link_info/ipvlan.rs:173]: Please rename it to `IpVlanFlags` as this is a bitflags.

File: src/link/tests/ipvlan.rs [hunk 3]
```diff
@@ -16,7 +16,7 @@ fn test_ipvlan_link_info() {
         0x00, 0x00, 0x00, 0x00, 0x24, 0x00, 0x12, 0x00, 0x0b, 0x00, 0x01, 0x00,
         0x69, 0x70, 0x76, 0x6c, 0x61, 0x6e, 0x00, 0x00, 0x14, 0x00, 0x02, 0x00,
         0x06, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06, 0x00, 0x02, 0x00,
-        0x02, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00,
```
Comment (cathay4t) [inline src/link/tests/ipvlan.rs:19]: Please do not modify captured data.

File: src/link/tests/ipvtap.rs [hunk 4]
```diff
@@ -16,7 +16,7 @@ fn test_ipvtap_link_info() {
         0x00, 0x00, 0x00, 0x00, 0x24, 0x00, 0x12, 0x00, 0x0b, 0x00, 0x01, 0x00,
         0x69, 0x70, 0x76, 0x74, 0x61, 0x70, 0x00, 0x00, 0x14, 0x00, 0x02, 0x00,
         0x06, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06, 0x00, 0x02, 0x00,
-        0x02, 0x00, 0x00, 0x00,
+        0x01, 0x00, 0x00, 0x00,
```
Comment (cathay4t) [inline src/link/tests/ipvtap.rs:19]: Please do not modify captured data.

If you want to test your enum value, please create new test case and do the nl_mon capture by following the README file of this repo.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: src/link/link_info/ipvlan.rs [hunk 5]
```diff
@@ -165,3 +163,24 @@ impl From<IpVlanMode> for u16 {
...
 }
+
+const IPVLAN_F_PRIVATE: u16 = 0x01;
+const IPVLAN_F_VEPA: u16 = 0x02;
+
+bitflags! {
+    #[non_exhaustive]
+    #[derive(Debug, Clone, Copy, Eq, PartialEq)]
+    pub struct IpVlanFlags: u16 {
+        const PRIVATE = IPVLAN_F_PRIVATE;
```
Comment (cathay4t) [inline src/link/link_info/ipvlan.rs:174]: Enum variants should be `UpperCamelCase`.
In this crate, bitflags created type are considered as enum.

**Review by cathay4t — CHANGES_REQUESTED** [review]

**Review by cathay4t — APPROVED** [review]
Nicely done! Thanks!

**Pull request merged**